### PR TITLE
214 donation status

### PR DESCRIPTION
**Pull request by Biplob14**
On every new row push on donation accept request, a lambda will trigger to check if required limit of blood donor is reached or not. If required number of donors are present on the row, then lambda will change the `status` of donation request to `completed`

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [x] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: core/application/bloodDonationWorkflow/BloodDonationService.ts [hunk 1]
```diff
@@ -160,7 +161,7 @@ export class BloodDonationService {
       }
 
       if (retryCount >= Number(process.env.MAX_RETRY_COUNT)) {
-        updateData.status = DonationStatus.EXPIRED
+        // updateData.status = DonationStatus.EXPIRED
```
Comment (EbrahimAli-RU) [inline core/application/bloodDonationWorkflow/BloodDonationService.ts:164]: Why commented code?

File: iac/terraform/aws/donor_search/sqs.tf [hunk 2]
```diff
@@ -54,3 +54,33 @@ data "aws_iam_policy_document" "donor_search_queue_policy_doc" {
     resources = [aws_sqs_queue.donor_search_queue.arn]
   }
 }
+
+
+# Queue for donation status manager
```
Comment (EbrahimAli-RU) [inline iac/terraform/aws/donor_search/sqs.tf:59]: ?

File: iac/terraform/aws/eventbridge/eventbridge.tf [hunk 3]
```diff
@@ -13,15 +13,11 @@ resource "aws_pipes_pipe" "donation_request_pipe" {
...
-                S : [{ prefix : "BLOOD_REQ#" }]
-              },
-              SK : {
-                S : [{ prefix : "BLOOD_REQ#" }]
-              }
+          "eventName" : ["INSERT", "MODIFY"],
+          "dynamodb" : {
+            "NewImage" : {
+              "PK" : { "S" : [{ "prefix" : "BLOOD_REQ#" }] },
+              "SK" : { "S" : [{ "prefix" : "BLOOD_REQ#" }] }
```
Comment (EbrahimAli-RU) [inline iac/terraform/aws/eventbridge/eventbridge.tf:20]: Is it possible to take these values `BLOOD_REQ` in the variable and then use it?

**Review by EbrahimAli-RU — CHANGES_REQUESTED** [review]

File: iac/terraform/aws/eventbridge/eventbridge.tf (hunk 3 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/eventbridge/eventbridge.tf:20]: we can. we will have to create a local variable. but it looks unnecessary.

File: core/application/bloodDonationWorkflow/BloodDonationService.ts (hunk 1 above)
Comment (mahinuzzaman) [reply core/application/bloodDonationWorkflow/BloodDonationService.ts]: Updated at 1655f5140d75497f04324c62cd8c93088ec99139

File: iac/terraform/aws/donor_search/sqs.tf (hunk 2 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/donor_search/sqs.tf]: Updated at 1655f5140d75497f04324c62cd8c93088ec99139

**Pull request merged**